addons/mail/models/mail_channel.py

```diff
@@ -92,6 +98,32 @@ def _compute_is_chat(self):
         for record in self:
             record.is_chat = record.channel_type == 'chat'
 
+    @api.depends('channel_last_seen_partner_ids.partner_id')
+    def _compute_channel_partner_ids(self):
+        for channel in self:
+            channel.channel_partner_ids = channel.channel_last_seen_partner_ids.partner_id
+
+    def _inverse_channel_partner_ids(self):
+        for channel in self:
+            current_members = channel.channel_last_seen_partner_ids
+            partners = channel.channel_partner_ids
+            partners_new = partners - current_members.partner_id
+
+            current_members.create([{
+                'channel_id': channel.id,
+                'partner_id': partner.id,
+            } for partner in partners_new])
+
+            current_members.filtered(lambda m: m.partner_id not in partners).unlink()
+
+    def _search_channel_partner_ids(self, operator, operand):
+        # TDE CHECKME: multicompany
+        memberships = self.env['mail.channel.partner'].sudo().search([
```

Comment:
I completely agree that we have an issue with naming. Issue is that channel_partners clashes with channel_partner_ids field. Membership is a term I used for team members, not sure if we have something else ... 

Ideally, we would have: channel_partners = res.partner, channel_members = mail.channel.member . Will maybe use channel_members temporarily as variable name until we update remaining fields. What do you think ?

Reply:
Well I would expect `channel_members` to be partners (or whatever new model we decide for mail user later), not a technical table...

I think the technical table between channel and its members is described perfectly by the `membership` word actually... It's just unfortunate we have a module named that way too.

`channel_membership` to remove the confusion?
And model renamed `mail.channel_membership` later?

Reply:
Well considering that membership is completely outdated, we may re-use its namespace. I bet people using membership will not yell at us. If they do, well, that does not change anything. Anyway when splitting channel into mailing lists / discord channels obviously it will be the time to update the naming :) .

Reply:
By the way, now using _search thingy.